official/vision/beta/projects/yolo/tasks/yolo.py

```diff
@@ -0,0 +1,442 @@
+        aug_rand_perspective=params.parser.aug_rand_perspective,
+        area_thresh=params.parser.area_thresh,
+        random_flip=params.parser.random_flip,
+        seed=params.seed,
+    )
+
+    # get the decoder
+    decoder = self.get_decoder(params)
+
+    # init Mosaic
+    sample_fn = mosaic.Mosaic(
+        output_size=model.input_size,
+        mosaic_frequency=params.parser.mosaic.mosaic_frequency,
+        mixup_frequency=params.parser.mosaic.mixup_frequency,
+        jitter=params.parser.mosaic.jitter,
+        mosaic_center=params.parser.mosaic.mosaic_center,
+        mosaic_crop_mode=params.parser.mosaic.mosaic_crop_mode,
+        aug_scale_min=params.parser.mosaic.aug_scale_min,
+        aug_scale_max=params.parser.mosaic.aug_scale_max,
+        **base_config)
+
+    # init Parser
+    parser = yolo_input.Parser(
+        output_size=model.input_size,
+        anchors=anchor_dict,
+        use_tie_breaker=params.parser.use_tie_breaker,
+        jitter=params.parser.jitter,
+        aug_scale_min=params.parser.aug_scale_min,
+        aug_scale_max=params.parser.aug_scale_max,
+        aug_rand_hue=params.parser.aug_rand_hue,
+        aug_rand_saturation=params.parser.aug_rand_saturation,
+        aug_rand_brightness=params.parser.aug_rand_brightness,
+        max_num_instances=params.parser.max_num_instances,
+        scale_xy=model.detection_generator.scale_xy.get(),
+        expanded_strides=model.detection_generator.path_scales.get(),
+        darknet=model.darknet_based_model,
+        best_match_only=params.parser.best_match_only,
+        anchor_t=params.parser.anchor_thresh,
+        random_pad=params.parser.random_pad,
+        level_limits=level_limits,
+        dtype=params.dtype,
+        **base_config)
+
+    # init the dataset reader
+    reader = input_reader.InputReader(
+        params,
+        dataset_fn=tf.data.TFRecordDataset,
+        decoder_fn=decoder.decode,
+        sample_fn=sample_fn.mosaic_fn(is_training=params.is_training),
+        parser_fn=parser.parse_fn(params.is_training))
+    dataset = reader.read(input_context=input_context)
+    return dataset
+
+  def build_metrics(self, training=True):
+    """Build detection metrics."""
+    metrics = []
+
+    backbone = self.task_config.model.backbone.get()
+    metric_names = collections.defaultdict(list)
+    for key in range(backbone.min_level, backbone.max_level + 1):
+      key = str(key)
+      metric_names[key].append('loss')
+      metric_names[key].append("avg_iou")
+      metric_names[key].append("avg_obj")
+
+    metric_names['net'].append('box')
+    metric_names['net'].append('class')
+    metric_names['net'].append('conf')
+
+    for i, key in enumerate(metric_names.keys()):
+      metrics.append(ListMetrics(metric_names[key], name=key))
+
+    self._metrics = metrics
+    if not training:
+      self.coco_metric = coco_evaluator.COCOEvaluator(
+          annotation_file=self.task_config.annotation_file,
```

Comment:
If annotation file is provided, the coco class mapping won't be correctly handled. Can you add a check to prevent eval from annotation_file if coco91_80 is in effect?